sdk/xmake.lua

```diff
@@ -39,6 +39,17 @@ debugOption("scheduler")
 debugOption("allocator")
 debugOption("token_library")
 
+function stackCheckOption(name)
+	option("stack-usage-check-" .. name)
+		set_default(false)
+		set_description("Enable dynamic stack usage checks in " .. name .. ". Do not enable this in debug builds!")
```

Comment:
Hm.  I'm not sure how I feel about the "do not enable in debug builds" bit.  I presume that's because the constants in the source are the release build stack usage and debug messages dramatically increase the stack usage.  Maybe debug builds should force any `Asserting` mode to be `Logging` instead?

Reply:
There's no value to running it in debug builds because they will have larger stack usage and so you'll bump your requirements to something wrong.  The only exceptions to that rule are people that really understand what they're doing and know that they can ignore the advice and, importantly, why.